src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfCie.cs

```diff
@@ -39,6 +39,19 @@ public DwarfCie(TargetArchitecture targetArchitecture)
 
             switch (targetArchitecture)
             {
+                case TargetArchitecture.ARM:
+                    CodeAlignFactor = 1;
+                    DataAlignFactor = -4;
+                    ReturnAddressRegister = 14; // LR
+                    Instructions = new byte[]
+                    {
+                        DW_CFA_def_cfa,
+                        13, // SP
+                        0, // Offset from SP
+                    };
+                    InitialCFAOffset = 0;
```

Comment:
There's multiple wrong values here. I'll switch back to draft and try to fix it first.

Reply:
The JIT was generating trash. Now it generates valid information but in extremely verbose way. On ARM64 we compress it in the managed code, or we could fix the JIT to generate it efficiently, but at least it's not a garbage and the unwinder can actually unwind all the frames up to the native code.

Reply:
FWIW, the code in `ObjectWriter::EmitARMExIdxPerOffset` in the old ObjWriter compensated for the missing `CFI_ADJUST_CFA_OFFSET` codes.

Reply:
Seems like the order of the registers on push is still wrong. :-/

Reply:
...and fixed the order now.